src/client/datascience/notebook/kernelWithMetadata.ts

```diff
@@ -63,6 +63,13 @@ export class VSCodeNotebookKernelMetadata implements VSCNotebookKernel {
      * resolved, the cell will be put back into the Idle state.
      */
     public async executeCellsRequest(document: NotebookDocument, ranges: NotebookCellRange[]): Promise<void> {
+        // When we receive a cell execute request, first ensure that the notebook is trusted.
+        // If it isn't already trusted, block execution until the user trusts it.
```

Comment:
Previously, one could not execute a cell of the notebook was not trusted.
Looks like something used now changes that's causing the run to be enabled.

I personally think that's a good thing. However I'd like  to understand what changed. Did something in vscode change?

Because right now we're not really fixing a big were fixing a regressing, and there's no choice change in here that indicates something broke...

Also would be good to have a test for this.

Reply:
Yeah, still draft status, been trying to get tests passing locally before I push :) 

I'm not entirely sure what changed to be honest. I imagine there could have been an unintentional change while refactoring, and we didn't have tests to enforce this specific scenario. 

> I personally think that's a good thing. 

FWIW I agree, we talked about this at triage and said the same, but it's still not working as designed. I'm also looking forward to being able to deprecate our trust support when VS Code releases trusted workspaces (I'm seeing their WIP API for trusted notebooks in the proposed API already)